database-factory/dbfactory.go

```diff
@@ -0,0 +1,49 @@
+package dbfactory
+
+import (
+	"fmt"
+	"strings"
+
+	terraform "github.com/mattermost/mattermost-cloud-database-factory/internal/tools/terraform"
+	"github.com/mattermost/mattermost-cloud-database-factory/model"
+	"github.com/pkg/errors"
+)
+
+var templateDir = "terraform/aws/database-factory"
+
+// CreateCluster is used to initiate Terraform and either Apply or Plan Terraform for the RDS Cluster deployments.
+func CreateCluster(cluster *model.Cluster) error {
+	logger.Info("Initialising Terraform")
+	stateObject := fmt.Sprintf("rds-cluster-multitenant-%s-%s", strings.Split(cluster.VPCID, "-")[1], cluster.ClusterID)
+
+	tf, err := terraform.New(templateDir, cluster.StateStore, logger)
+	if err != nil {
+		logger.WithError(err).Error("failed to initiate Terraform")
```

Comment:
I see that we are logging every error manually because we were originally ignoring the returned error value on the return function. It would be nice to be able to log them all in one place like I mentioned.

Reply:
Yeap makes total sense. I fixed in the comment above ;)